### ISSUE TITLE
Settings not being saved

### ISSUE THREAD

**Issue by paulnagle** [bmlt-enabled/wordpress-bmlt-workflow#65]
Hello! I set up a new instance of wordpress with this plugin. I filled in all of the "Configuration" details and Clicked on "Save Changes"

I see the following errors (I have WP_DEBUG enabled

```
Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'string_sanitize_callback' not found or invalid function name in /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php on line 309

Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'string_sanitize_callback' not found or invalid function name in /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php on line 309

Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'string_sanitize_callback' not found or invalid function name in /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php on line 309

Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'string_sanitize_callback' not found or invalid function name in /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php on line 309

Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'string_sanitize_callback' not found or invalid function name in /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php on line 309

Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'string_sanitize_callback' not found or invalid function name in /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php on line 309

Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'string_sanitize_callback' not found or invalid function name in /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php on line 309

Warning: Cannot modify header information - headers already sent by (output started at /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php:309) in /home/nasouth0/updates.nasouth.ie/wp-includes/pluggable.php on line 1421

Warning: Cannot modify header information - headers already sent by (output started at /home/nasouth0/updates.nasouth.ie/wp-includes/class-wp-hook.php:309) in /home/nasouth0/updates.nasouth.ie/wp-includes/pluggable.php on line 1424
```

When I go back to the "Configuration" page, I see that none of my changes have stuck, templates for emails are blank etc.

I can see that the "BMLT Configuration" does seem to get saved, and the "Service Bodies" section does retain it's settings.

**Comment by paulnagle** [comment]
PS. I was using the following docker-compose.yaml file to test

```
version: "3.9"

services:
  db:
    image: mysql:5.7
    volumes:
      - db_data:/var/lib/mysql
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: somewordpress
      MYSQL_DATABASE: wordpress
      MYSQL_USER: wordpress
      MYSQL_PASSWORD: wordpress

  wordpress:
    depends_on:
      - db
    image: wordpress:latest
    volumes:
      - wordpress_data:/var/www/html
      - ../bmlt_tabbed_map/:/var/www/html/wp-content/plugins/bmlt_tabbed_map:ro
    ports:
      - "8000:80"
    restart: always
    environment:
      WORDPRESS_DB_HOST: db
      WORDPRESS_DB_USER: wordpress
      WORDPRESS_DB_PASSWORD: wordpress
      WORDPRESS_DB_NAME: wordpress
volumes:
  db_data: {}
  wordpress_data: {}
```

and justr running `docker-compose up -d` from the git repo folder to start the wp instance .

**Comment by nigel-bmlt** [comment]
fixed in 0.4.2

**Comment by nigel-bmlt** [comment]
thanks for the debug output - this is a super helpful report :)
